include/sparrow/fixed_size_layout.hpp

```diff
@@ -135,7 +140,8 @@ namespace sparrow
         const_value_range values() const;
 
     private:
-        array_data m_data;
+
+        array_data* p_data;
```

Comment:
Why storing the pointer and not the reference ?

Reply:
It's more flexible:
- the layout can become easily default constructible if we need it
- we can assign to a pointer, not to a reference
- this storage is private, so it's fine to use a pointer instead of a reference..

Reply:
What about using std::reference_wrapper ?
My point is that a reference explicits the ownership. It's less clear with a pointer.

Reply:
That's a good point. Let me update the PR.